vela-templates/definitions/api-oriented/component/webservice.cue

```diff
@@ -0,0 +1,260 @@
+								{
+									mountPath: v.mountPath
+									name:      v.name
+								}}]
+						}
+
+						if parameter["livenessProbe"] != _|_ {
+							livenessProbe: parameter.livenessProbe
+						}
+
+						if parameter["readinessProbe"] != _|_ {
+							readinessProbe: parameter.readinessProbe
+						}
+
+					}]
+
+					if parameter["imagePullSecrets"] != _|_ {
+						imagePullSecrets: [ for v in parameter.imagePullSecrets {
+							name: v
+						},
+						]
+					}
+
+					if parameter["volumes"] != _|_ {
+						volumes: [ for v in parameter.volumes {
+							{
+								name: v.name
+								if v.type == "pvc" {
+									persistentVolumeClaim: claimName: v.claimName
+								}
+								if v.type == "configMap" {
+									configMap: {
+										defaultMode: v.defaultMode
+										name:        v.cmName
+										if v.items != _|_ {
+											items: v.items
+										}
+									}
+								}
+								if v.type == "secret" {
+									secret: {
+										defaultMode: v.defaultMode
+										secretName:  v.secretName
+										if v.items != _|_ {
+											items: v.items
+										}
+									}
+								}
+								if v.type == "emptyDir" {
+									emptyDir: medium: v.medium
+								}
+							}}]
+					}
+				}
+			}
+		}
+	}
+	parameter: {
+		// +usage=Which image would you like to use for your service
+		// +short=i
+		image: string
+
+		// +usage=Specify image pull policy for your service
+		imagePullPolicy?: string
+
+		// +usage=Specify image pull secrets for your service
+		imagePullSecrets?: [...string]
+
+		// +usage=Which ports do you want customer traffic sent to, defaults to 80
+		// +short=p
+		ports?: [...int]
+
+		// +ignore
+		// +usage=If addRevisionLabel is true, the appRevision label will be added to the underlying pods
+		addRevisionLabel: *false | bool
+
+		// +usage=Commands to run in the container
+		cmd?: [...string]
+
+		// +usage=Define arguments by using environment variables
+		env?: [...{
+			// +usage=Environment variable name
+			name: string
+			// +usage=The value of the environment variable
+			value?: string
+			// +usage=Specifies a source the value of this var should come from
+			valueFrom?: {
+				// +usage=Selects a key of a secret in the pod's namespace
+				secretKeyRef: {
+					// +usage=The name of the secret in the pod's namespace to select from
+					name: string
+					// +usage=The key of the secret to select from. Must be a valid secret key
+					key: string
+				}
+			}
+		}]
+
+		// +usage=Number of CPU units for the service, like `0.5` (0.5 CPU core), `1` (1 CPU core)
+		cpu?: string
+
+		// +usage=Specifies the attributes of the memory resource required for the container.
+		memory?: string
+
+		// +usage=Declare volumes and volumeMounts
+		volumes?: [...{
+			name:      string
+			mountPath: string
+			// +usage=Specify volume type, options: "pvc","configMap","secret","emptyDir"
+			type: "pvc" | "configMap" | "secret" | "emptyDir"
+			if type == "pvc" {
+				claimName: string
```

Comment:
Can I directly specify the storage class to generate PVC?

Reply:
Should we generate PVC in component but not trait?

Reply:
The binding is required, which means that it needs to be created before that. How does a trait achieve this effect?

Reply:
If you need to create the storage and mount it, use `storage` trait instead of specify in the component.

Reply:
ref: https://github.com/oam-dev/kubevela/pull/2664/files#diff-93b6b5be1dfdc598a25430a40d40422f68aafee6bf19ae73f1021efa0b0cefc5R148